graphql/src/index.ts

```diff
@@ -23,47 +22,71 @@ const { ElasticSearchAPI } = require('./datasources/es');
+  ) {
+    return {
+      cursor: getCursor(index + 1),
+      node: {
+        _score: e._score,
+        venue: { venue: e._source.venue }, // pass parent to child resolver. How to do this better?
+      },
+    };
+  });
+}
+
+function getHits(results: any) {
+  return results.hits.total.value;
+}
+
 const resolvers = {
   Query: {
-    unifiedSearch: async (_source: any, { q, ontology, index }: any, { dataSources }: any) => {
-      const res = await dataSources.elasticSearchAPI.getQueryResults(q, ontology, index);
-      return { es_results: res };
+    unifiedSearch: async (
+      _source: any,
+      { q, ontology, index, before, after, first, last }: UnifiedSearchQuery,
+      { dataSources }: any
+    ) => {
+      const connectionArguments = { before, after, first, last };
+
+      const res = await dataSources.elasticSearchAPI.getQueryResults(
+        q,
+        ontology,
+        index,
+        connectionArguments
+      );
+      const results = await res[0];
+
+      const getCursor = (offset: number) =>
+        createCursor<ConnectionCursorObject>({
+          offset,
+        });
+
+      // Find shared data
+      const edges = edgesFromEsResults(results, getCursor);
+      const hits = getHits(results);
+
+      return { es_results: results, edges, hits, connectionArguments };
     },
   },
   SearchResultConnection: {
-    count(parent: { es_results: any; }, args: any, context: any, info: any) {
-      const { es_results } = parent;
-      return es_results[0].then((r: { hits: { total: { value: any; }; }; }) => r.hits.total.value);
+    count({ hits }: any) {
+      return hits;
     },
-    max_score(parent: { es_results: any; }, args: any, context: any, info: any) {
-      const { es_results } = parent;
-      return es_results[0].then((r: { hits: { max_score: any; }; }) => r.hits.max_score);
+    max_score({ es_results }: any) {
+      return es_results.hits.max_score;
     },
-    pageInfo(parent: any, args: any, context: any, info: any) {
-      return {
-        hasNextPage: false,
-        hasPreviousPage: false,
-        startCursor: 'startCursor123',
-        endCursor: 'endCursor123',
-      };
+    pageInfo({ edges, hits, connectionArguments }: any) {
+      return pageInfoResolver(edges, hits, connectionArguments);
     },
-    edges(parent: { es_results: any; }, args: any, context: any, info: any) {
-      console.log('at edges');
-      const { es_results } = parent;
-      console.log(es_results);
-
-      es_results[0].then((r: any) => console.log(r));
-      const edges = es_results[0].then((r: { hits: { hits: any[]; }; }) =>
-        r.hits.hits.map(function (e: { _score: any; _source: { venue: any; }; }) {
-          return {
-            cursor: 123,
-            node: {
-              _score: e._score,
-              venue: { venue: e._source.venue }, // pass parent to child resolver. How to do this better?
-            },
-          };
-        })
-      );
+    edges({ edges }: any) {
```

Comment:
I removed quite a bit of logging here: I was not sure if it was pertinent or not. If they were important, I can add them back if you are able to give me some sense of the prupose.